recipe/meta.yaml

```diff
@@ -40,41 +43,41 @@ requirements:
     - python
     - pip
     - cython
-    - arrow-cpp
+    - arrow-cpp >=8,<9
     - pyarrow
     - numpy
     - setuptools >=40.6.0
     - wheel
   run:
     - python
-    - {{ pin_compatible('pyarrow', max_pin='x') }}
-    # While requests is vendored, we use regular requests to perform OCSP checks
-    - requests <3.0.0
-    - pytz
-    - pycryptodomex >=3.2,!=3.5.0,<4.0.0
-    - pyopenssl >=16.2.0,<23.0.0
+    - pyarrow
+    - {{ pin_compatible('arrow-cpp', max_pin='x') }}
+    - asn1crypto >0.24.0,<2.0.0
     - cffi >=1.9,<2.0.0
     - cryptography >=3.1.0,<37.0.0
-    - pyjwt <3.0.0
     - oscrypto <2.0.0
-    - asn1crypto >0.24.0,<2.0.0
-    # A functioning pkg_resources.working_set.by_key and pkg_resources.Requirement is
-    # required. Python 3.6 was released at the end of 2016. setuptools 34.0.0 was released
-    # in early 2017, so we pick this version as a reasonably modern base.
+    - pyopenssl >=16.2.0,<23.0.0
+    - pycryptodomex >=3.2,!=3.5.0,<4.0.0
+    - pyjwt <3.0.0
+    - pytz
+    - requests <3.0.0
     - setuptools >34.0.0
-    # requests requirements
-    - charset-normalizer ~=2.0.0
+    - charset-normalizer >=2,<3
     - idna >=2.5,<4
+    - urllib3 >=1.21.1,<1.27
     - certifi >=2017.4.17
+    - typing-extensions >=4.3.0,<5
   run_constrained:
     - pandas >1,<1.5
+    - keyring !=16.1.0,<24.0.0
 
 test:
   requires:
     - pip
   imports:
     - snowflake
     - snowflake.connector
+    - snowflake.connector.arrow_iterator
```

Comment:
This ensures that the linking between snowflake-connector-python and arrow-cpp is successful. I confirmed this approach with the developer.